tests/requirements_file_exercise_tests.py

```diff
@@ -0,0 +1,132 @@
+    def test_should_create_libraries_string_from_single_file(self, mock_read_file_lines):
+        mock_read_file_lines.return_value = ["Flask==2.0.2", "Flask-RESTful==0.3.9", "Jinja2==3.0.2",
+                                             "validators==0.18.2"]
+        file_name = 'path_to_rquirements_file.txt'
+        excepted = "Flask Flask-RESTful Jinja2 validators"
+
+        actual = create_libraries_string(file_name)
+
+        mock_read_file_lines.assert_called_once_with(file_name)
+        self.assertEqual(excepted, actual)
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_should_create_libraries_string_from_file_with_references(self, mock_read_file_lines):
+        referenced_requirement_file = 'test-requirements.txt'
+        mock_lines_of_first_requirement_file = ["Flask==2.0.2", "Flask-RESTful==0.3.9", "Jinja2==3.0.2",
+                                                "-r " + referenced_requirement_file, "validators==0.18.2"]
+        mock_lines_of_second_requirement_file = ["requests==2.26.0", "six==1.16.0", "typing-extensions==3.10.0.2"]
+        mock_read_file_lines.side_effect = [mock_lines_of_first_requirement_file, mock_lines_of_second_requirement_file]
+        main_requirements_file = 'requirements_file.txt'
+        excepted = "Flask Flask-RESTful Jinja2 validators requests six typing-extensions"
+
+        actual = create_libraries_string(main_requirements_file)
+
+        mock_read_file_lines.assert_has_calls([call(main_requirements_file), call(referenced_requirement_file)])
+        self.assertEqual(excepted, actual)
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_create_libraries_string_should_add_just_one_instance_of_library_to_list(self, mock_read_file_lines):
+        mock_read_file_lines.return_value = ["Flask==2.0.2", "Flask-RESTful==0.3.9", "Jinja2==3.0.2",
+                                             "validators==0.18.2", "Flask==2.0.2", "Jinja2==3.0.2"]
+        file_name = 'path_to_rquirements_file.txt'
+        excepted = "Flask Flask-RESTful Jinja2 validators"
+
+        actual = create_libraries_string(file_name)
+
+        mock_read_file_lines.assert_called_once_with(file_name)
+        self.assertEqual(excepted, actual)
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_should_create_empty_string_for_empty_file(self, mock_read_file_lines):
+        excepted = ""
+        mock_read_file_lines.return_value = []
+        file_name = 'path_to_rquirements_file.txt'
+
+        actual = create_libraries_string(file_name)
+
+        mock_read_file_lines.assert_called_once_with(file_name)
+        self.assertEqual(excepted, actual)
+
+    @mock.patch('src.requirements_file_exercise.read_file_lines')
+    def test_should_read_once_file_that_has_more_than_one_reference(self, mock_read_file_lines):
+        first_requirements_file = 'requirements.txt'
+        second_requirements_file = 'second-requirements.txt'
```

Comment:
here you did it right :) In contrast to the above 'def test_should_create_libraries_string_from_file_with_references'